### DISCUSSION TITLE
Is there Api to fetch chart data (with filter applied) from dashboard?

### DISCUSSION THREAD

**Discussion by mtv2102** [apache/superset#21757] [Q&A / Help] [answered]
I have tried using api such as 'api/v1/chart/data?format=json' to fetch chart data in json format. But when I try to fetch the chart data with filter applied on dashboard, I couldn't  find any api so far.

I have checked with the following url..http://localhost:8088/swagger/v1 . It has listed the set of supported api's.

Among that i have found something like,  `api/v1/dashboard/14(dashboard_id)/charts `

But I couldn't find any api to get the filtered chart data ...

Filter Applied
<img width="194" alt="image" src="https://user-images.githubusercontent.com/112002949/194867660-ee15d1ef-0809-4a47-8393-8f0f9475b337.png">

Chart Data with applied filter in dashboard
<img width="380" alt="image" src="https://user-images.githubusercontent.com/112002949/194867743-b52db6a7-81ea-475e-b73f-93591bdfeb87.png">

I have spent my entire day on trying this, but couldn't find.

Any help / guidance would be appreciated.

**Comment by mayurnewase** [comment, marked as answer]
payload for that api is generated by the frontend according to filters present in dashboard / chart, so you have get the chart / dashboard filter first then construct payload and send to this api.

**Reply by mtv2102** [reply]
@mayurnewase Thanks for the info. I have found the logic for that and attached the code snippet for that.
 ```
  await getChartDataRequest({
          formData: this.props.formData,
          resultFormat: 'json',
          resultType:'results',
        })
          .then(({ json }) => {}));
```
The above code actually fetches the filtered data in json format :))

**Comment by mayurnewase** [comment]
also this question is better suited in discussion.

**Reply by Gokuyamatos12** [reply]
I'm actually trying to use custom react filter components to pass values to my embedded superset dashboard. Since the SDK doesn't let me pass filter values and creating URL filter parameters make the entire dashboard reload (ruining user experience), I've been trying to send the filter values through a post request. But I don't know the API endpoint to update the dashboard charts with filter values that I want passed, is there an endpoint? or do I have to create a custom endpoint?

**Reply by avinashprj** [reply]
Same, is there any way to change filter data via the superset api?

**Reply by bryan-oliveira** [reply]
Running into same issue here. We have charts embedded in our app, and would like to apply filters to the embedded chart.

**Comment by r3dge** [comment]
Can anyone share an exemple of API queries to show how to apply a dashboard filter ? Help much appreciated.
